purchase_discount/tests/test_product_supplierinfo_discount.py

```diff
@@ -30,10 +30,10 @@ def setUpClass(cls):
             {
                 "min_qty": 10.0,
                 "partner_id": cls.partner_3.id,
-                "product_tmpl_id": cls.product.product_tmpl_id.id,
                 "discount": 20,
             }
         )
+        cls.supplierinfo2["product_tmpl_id"] = cls.product.product_tmpl_id
```

Comment:
Put a comment why this weird thing:

```suggestion
        # Assign it later this value for testing that the record
        # creation is not failing without this field
        cls.supplierinfo2.product_tmpl_id = cls.product.product_tmpl_id
```

Reply:
Thanks, fixed.